packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,191 @@
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+        };
+    }>;
+    disconnect(): Promise<{
+        msg: string;
+    }>;
+    signTransaction(message: string): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signature: string;
+        }
+    }>;
+    signAllTransactions(messages: string[]): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+            signatures: string[];
+        }
+    }>;
+}
+
+interface SlopeWindow extends Window {
+    Slope?: any;
+}
+
+declare const window: SlopeWindow;
+
+export interface SlopeWalletAdapterConfig {
+    pollInterval?: number;
+    pollCount?: number;
+}
+
+export class SlopeWalletAdapter extends BaseSignerWalletAdapter {
+    private _connecting: boolean;
+    private _wallet: SlopeWallet | null;
+    private _publicKey: PublicKey | null;
+
+    constructor(config: SlopeWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (!this.ready) pollUntilReady(this, config.pollInterval || 1000, config.pollCount || 3);
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get ready(): boolean {
+        return !!window.Slope;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._publicKey;
+    }
+
+    get autoApprove(): boolean {
+        return false;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            this._connecting = true;
+
+            if (!window.Slope) throw new WalletNotFoundError();
+
+            const wallet = this._wallet || new window.Slope();
+            if (!wallet.connect) throw new WalletNotInstalledError()
+
+            let account: string;
+            try {
+               const { msg, data } = await wallet.connect();
+
+               if (msg !== 'ok') {
+                   throw new WalletAccountError(msg);
+               }
+
+               account = data.publicKey;
+            } catch (error) {
+                throw new WalletAccountError(error?.message, error);
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(account);
+            } catch (error) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.isConnected = true;
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect');
+        } catch (error) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        if (this._wallet) {
+            this._wallet.disconnect();
```

Comment:
`_wallet.disconnect` returns a Promise and this function is async so it should await that Promise and catch any errors. Please see a reference implementation @ https://github.com/solana-labs/wallet-adapter/blob/f14d5b30f7bdd2f47ee03c0744298f44b8803210/packages/phantom/src/adapter.ts#L157-L161

Reply:
Modified